packages/maskbook/src/plugins/Wallet/UI/ConnectWalletDialog/index.tsx

```diff
@@ -0,0 +1,136 @@
+
+export interface ConnectWalletDialogProps {}
+
+export function ConnectWalletDialog(props: ConnectWalletDialogProps) {
+    const { t } = useI18N()
+    const classes = useStylesExtends(useStyles(), props)
+
+    const account = useAccount()
+    const chainId = useChainId()
+    const [providerType, setProviderType] = useState<ProviderType | undefined>()
+
+    const selectedNetworkType = useValueRef(currentSelectedWalletNetworkSettings)
+    const selectedProviderType = useValueRef(currentSelectedWalletProviderSettings)
+
+    //#region remote controlled dialog
+    const { open, closeDialog } = useRemoteControlledDialog(WalletMessages.events.connectWalletDialogUpdated, (ev) => {
+        if (!ev.open) return
+        setProviderType(ev.providerType)
+    })
+    //#endregion
+
+    //#region wallet status dialog
+    const { openDialog: openWalletStatusDialog } = useRemoteControlledDialog(
+        WalletMessages.events.walletStatusDialogUpdated,
+    )
+    //#endregion
+
+    //#region walletconnect
+    const { setDialog: setWalletConnectDialog } = useRemoteControlledDialog(
+        WalletMessages.events.walletConnectQRCodeDialogUpdated,
+    )
+    //#endregion
+
+    const connectToMetamask = useCallback(async () => {
+        try {
+            const account = await Services.Ethereum.connectMetaMask()
+            if (!account) throw new Error('Failed to connect MetaMask.')
+
+            // read the chain detailed from the built-in chain list
+            const chainDetailed = CHAINS.find((x) => x.chainId === resolveNetworkChainId(selectedNetworkType))
+            if (!chainDetailed) throw new Error('The selected network is not supported.')
+
+            // it's unable to send a request for switching to ethereum networks
+            if (selectedNetworkType === NetworkType.Ethereum) {
+                if (chainDetailed.chain === 'ETH') return true
+                else throw new Error('Make sure your wallet is on the Ethereum network.')
+            }
+
+            // request ethereum-compatiable network
+            await Services.Ethereum.addEthereumChain(account, {
+                chainId: `0x${chainDetailed.chainId.toString(16)}`,
+                chainName: chainDetailed.name,
+                nativeCurrency: chainDetailed.nativeCurrency,
+                rpcUrls: chainDetailed.rpc,
+                blockExplorerUrls: [
+                    chainDetailed.explorers && chainDetailed.explorers.length > 0 && chainDetailed.explorers[0].url
+                        ? chainDetailed.explorers[0].url
+                        : chainDetailed.infoURL,
+                ],
+            })
+            return true as const
+        } catch (e) {
+            throw new Error(e.message)
+        }
+    }, [account, chainId, selectedNetworkType, selectedProviderType])
+
+    const connectToWalletConnect = useCallback(async () => {
+        const [uri_] = await Promise.allSettled([await Services.Ethereum.createConnectionURI(), delay(1000)])
```

Comment:
The 1000 looks like a magic number. 👀

Reply:
I think it's kind of waiting at least 1s to reduce flicker.